sky/data/sky_csync.py

```diff
@@ -0,0 +1,545 @@
+    sys.stdout.flush()
+    while True:
+        start_time = time.time()
+        run_sync(csync_write_path, storetype, destination, num_threads,
+                 interval_seconds, delete, no_follow_symlinks, csync_pid)
+        end_time = time.time()
+        # Given the interval_seconds and the time elapsed during the sync
+        # operation, we compute remaining time to wait before the next
+        # sync operation.
+        elapsed_time = int(end_time - start_time)
+        remaining_interval = max(_MIN_SYNC_CALL_INTERVAL,
+                                 interval_seconds - elapsed_time)
+        # sync_pid column is set to 0 when sync is not running
+        time.sleep(remaining_interval)
+
+
+@main.command()
+@click.argument('paths', nargs=-1, required=False)
+@click.option('--all',
+              '-a',
+              default=False,
+              is_flag=True,
+              required=False,
+              help='Terminates all CSYNC processes.')
+def terminate(paths: List[str], all: bool = False) -> None:  # pylint: disable=redefined-builtin
+    """Terminates the CSYNC daemon running.
+
+    Args:
+        paths (List[str]): list of CSYNC-mounted paths
+        all (bool): determine either or not to unmount every CSYNC-mounted
+            paths
+
+    Raises:
+        click.UsageError: when the paths are not specified
+    """
+    if not paths and not all:
+        raise click.UsageError('Please provide the CSYNC-mounted path to '
+                               'terminate the CSYNC process.')
+    _terminate(paths, all)
+
+
+def _terminate(paths: List[str], all: bool = False) -> int:  # pylint: disable=redefined-builtin
+    """Terminates the CSYNC daemon running.
+
+    Before terminating the running CSYNC daemon, it checks if the sync process
+    spawned by the daemon is running. If it is, we wait until the sync gets
+    completed, and then proceed to terminate the daemon.
+    """
+    csync_pid_set = set()
+    if all:
+        for csync_pid in _get_all_running_csync_pid():
+            csync_pid_set.add(csync_pid)
+    # when terminating specified CSYNC processes given the paths where
+    # the processes are running on.
+    else:
+        for path in paths:
+            full_path = os.path.abspath(os.path.expanduser(path))
+            csync_pid = _get_csync_pid_from_mountpoint_path(full_path)
+            if csync_pid is not None:
+                csync_pid_set.add(csync_pid)
+
+    pid_set = set()
+    for csync_pid in csync_pid_set:
+        storage_mount_pid, redirect_mount_pid = (
+            _get_storage_and_redirect_mount_pid(csync_pid))
+        pid_set.add((csync_pid, storage_mount_pid, redirect_mount_pid))
+
+    failed_to_terminate: bool = False
+    failed_to_terminate_list: List[Tuple[int, str, str]] = []
+    unmount_cmd = 'sudo fusermount -uz {unmount_path}'
+    for csync_pid, storage_mount_pid, redirect_mount_pid in pid_set:
+        # 1.unmount cloud storage from read path
```

Comment:
```suggestion
        # 1. unmount cloud storage from read path
```
nit; same for other comments